src/Form/atoms/ButtonWithIcon.tsx

```diff
@@ -38,37 +39,34 @@ const IconContainer = styled.div<{ position?: string }>`
   }
 `;
 
-const InnerContainer = styled.div<{position?: string}>`
+const InnerContainer = styled.div<{size?:TypeButtonSizes }>`
   display: flex;
   height: inherit;
-
-
-  ${({ position }) => position && position === 'left' ? css`
-    margin-right: -20px;
-  ` : css`
-    margin-left: -20px;
-  `}
-
+  ${({size}) => (size==='extraSmall')? css`margin-top: -3px` : ``};
+  ${({size}) => (size==='extraSmall')? css`padding-right: 10px` : ``};
+  margin-left: ${({size}) => (size==='extraSmall') ? `-10px` : `-20px` };
 `;
 
 interface IProps extends IButtonProps {
   icon: string
   position?: 'left' | 'right'
 }
 
-const ButtonWithIcon : React.FC<IProps> = ({design, size, onClick, disabled, position, icon, children, ...props}) => {
+const ButtonWithIcon : React.FC<IProps> = ({design, size = 'normal', onClick, disabled, position, icon, children, ...props}) => {
 
   const iconSize = size === 'large' ? 20 : 16;
```

Comment:
We need to make the icon (and for the loading version too) smaller for the xSmall size. I think `12` is what we want for that. Perhaps this can be refactored and added to the theme dimensions file too.